File: 백준/Silver/10828. 스택/스택.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();

        Stack<Integer> stack = new Stack<>();

        int N = sc.nextInt();

        for(int i = 0; i < N; i++) {
            String str = sc.next();

            switch (str) {
                case "push":
                    stack.push(sc.nextInt());
                    break;

                case "pop":
                    if (stack.isEmpty()) {
                        sb.append(-1);
                    } else {
                        sb.append(stack.pop());
                    }
                    sb.append("\n");
                    break;

                case "size":
                    sb.append(stack.size()).append("\n");
                    break;

                case "empty":
                    if (stack.isEmpty()) {
                        sb.append(1);
                    } else {
                        sb.append(0);
                    }
                    sb.append("\n");
                    break;

                case "top":
                    if (stack.isEmpty()) {
                        sb.append(-1);
                    } else {
                        sb.append(stack.peek());
                    }
                    sb.append("\n");
                    break;
            }
        }
        sc.close();
        System.out.println(sb);
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.util.*
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val N = readLine().toInt()
    val stack = Stack<Int>()
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    repeat(N) {
        val st = StringTokenizer(readLine())
        when(st.nextToken()) {
            "push" -> stack.push(st.nextToken().toInt())
            "pop" -> if(stack.isEmpty()) bw.write("-1\n") else bw.write("${stack.pop()}\n")
            "size" -> bw.write("${stack.size}\n")
            "empty" -> if(stack.isEmpty()) bw.write("1\n") else bw.write("0\n")
            "top" -> if(stack.isEmpty()) bw.write("-1\n") else bw.write("${stack.peek()}\n")
        }
    }
    close()
    bw.flush()
    bw.close()
}